pkgs/tools/security/tracee/default.nix

```diff
@@ -0,0 +1,109 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+
+, llvmPackages_13
+, pkg-config
+
+, zlib
+, libelf
+}:
+
+let
+  inherit (llvmPackages_13) clang;
+  clang-with-bpf =
+    (clang.overrideAttrs (o: { pname = o.pname + "-with-bpf"; })).override (o: {
+      extraBuildCommands = o.extraBuildCommands + ''
+        # make a separate wrapped clang we can target at bpf
+        cp $out/bin/clang $out/bin/clang-bpf
+        # extra flags to append after the cc-cflags
+        echo '-target bpf -fno-stack-protector' > $out/nix-support/cc-cflags-bpf
+        # use sed to attach the cc-cflags-bpf after cc-cflags
+        sed -i -E "s@^(extraAfter=\(\\$\NIX_CFLAGS_COMPILE_.*)(\))\$@\1 $(cat $out/nix-support/cc-cflags-bpf)\2@" $out/bin/clang-bpf
+      '';
+    });
+in
+buildGoModule rec {
+  pname = "tracee";
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "aquasecurity";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-Y++FWxADnj1W5S3VrAlJAnotFYb6biCPJ6dpQ0Nin8o=";
+    fetchSubmodules = true;
```

Comment:
Can't we use our libbpf?

Reply:
I think I got it to work but I was thinking it might be best if the libbpf version was updated in tandem with tracee.

I guess we could use our libbpf until it breaks.
I might write a nixos test to make sure it stays working.
A normal test probably wont play ball since itll need sudo and can probably trace outside the sandbox

Reply:
Ok I tried using `libbpf.a` directly from nixpkgs but I couldn't really get it working

I also tried just copying the `libbpf.src` into `3rdparty/libbpf` but it wasn't very happy about that either

I think part of the issue is the whole `$(OUT_DIR)/whatever:` stuff but to be fair I'm not an avid make user

Reply:
Please don't use your own libbpf. We test features per libbpf release (and per libbpfgo release). It is better if you use our builtin version just like we do in our packaging.

Reply:
Cool, thanks for the review

Reply:
You are [pinning libbpf version 0.7.0](https://github.com/aquasecurity/tracee/tree/v0.7.0/3rdparty) which is the exact version we currently have. There are no patches and vendoring in this case has only disadvantages for us: We rely on upstream to update it and patches from our libbpf are not applied. It might break when other libraries are updated and wouldn't be fixed. Also vendored libraries increase compile times because we build the lib more often than necessary.

Reply:
@SuperSandro2000 I know about the cons of having a static lib redistributed and that is not something I would request from any other project usually (I'm an Ubuntu core dev). Unfortunately, libbpf isn't stable until 1.0 (they try to be, but there might be breaking changes until they're fully stable) AND we have faced issues from 0.5 to 0.7, for example. So, let's say next release (0.7.1, or 0.8) of tracee still depends on libbpf0.7 + libbpfgo0.7+xxx and you may need to update libbpf to 0.8 to follow upstream, because of others tools (like bpftool), then we would start having a problem. If your users come to our github to open an issue we would request them to drop your package and use ours (instead of recommending them to use yours like we could do). At the end, it is your decision obviously, it was just a kind request. Thanks!

Reply:
Ok, these are the two approaches Im happy with:

1. Keep it as is at least until libbpf hits 1.0
2. Use the nixpkgs libbpf but I'll put some checks to make sure it matches the pinned version tracee wants. If the main nixpkgs libbpf goes ahead Ill pin a separate one at the right version

My preference is for option 1 since its less overhead and a minor compromise for a higher guarantee of tracee working.

Option 2 is quite a lot of work and at that point Id probably go with an option 3 which would be - just use the tracee release bin like the first draft of this PR, where we'd go from building 90% ideal solution to a less ideal one

Reply:
I like the approach of having static libbpf for tracee until libbpf is 1.0. I think that, from point on, if we find an issue in tracee because of libbpf then it will be on us either way. Thanks for considering it.

Reply:
yeah, fine with me.